### PR TITLE
7048 gas est reporting

### PR DESCRIPTION
**Pull request by bthaile**
create separate call to get gas estimates to display to initial reporter. convert gas to gas costs using constants, round to 4 decimals based on UI mocks

https://app.clubhouse.io/augur/story/7048/add-gas-costs-to-reporting-report-confirm-form

### CODE REVIEW

File: src/modules/reporting/actions/submit-initial-report.js [hunk 1]
```diff
@@ -28,3 +29,29 @@ export const submitInitialReport = (marketId, selectedOutcome, invalid, history,
     })
   }
 }
+
+
+export const estimateSubmitInitialReport = (marketId, callback = logError) => (dispatch, getState) => {
```
Comment (justinbarry) [inline src/modules/reporting/actions/submit-initial-report.js]: Please add test coverage

File: src/modules/reporting/components/reporting-report/reporting-report.jsx [hunk 2]
```diff
@@ -88,8 +92,21 @@ export default class ReportingReport extends Component {
         reporterGasCost: formatEtherEstimate(marketCreationCostBreakdown.targetReporterGasCosts),
         stake: this.props.isOpenReporting ? '0' : repAmount.formatted,
       })
+    })
+  }
+
+  calculateGasEstimates() {
```
Comment (justinbarry) [inline src/modules/reporting/components/reporting-report/reporting-report.jsx]: Please breakout out into a util file and add test coverage.

File: src/modules/reporting/actions/submit-initial-report.js (hunk 1 above)
Comment (bthaile) [reply src/modules/reporting/actions/submit-initial-report.js]: It simple server call no logic to test

File: src/modules/reporting/components/reporting-report/reporting-report.jsx (hunk 2 above)
Comment (bthaile) [reply src/modules/reporting/components/reporting-report/reporting-report.jsx]: ok, will break out.

File: src/modules/reporting/actions/submit-initial-report.js (hunk 1 above)
Comment (bthaile) [reply src/modules/reporting/actions/submit-initial-report.js]: ok, will write unit test for logic based on service call success or fails. makes sense

Comment (bthaile) [reply src/modules/reporting/actions/submit-initial-report.js]: done unit tests written

**Comment by bthaile** [comment]
@justinbarry 
are you suggesting breaking out the `calculateGasEstimates` method out of the component. I broke out formatting gas cost to ether into a help method with unit tests.

**Comment by justinbarry** [comment]
@bthaile Maybe? Breaking out the formatter like you did is good enough for me. 

If we have a ban on  async inside components then maybe it should move but this is something we should decide as a team. Balancing the practical with the fancy is hard sometimes ;)

**Comment by bthaile** [comment]
resolved merge conflict should be good to go.

**Comment by justinbarry** [comment]
Looks good!

screen shot 2018-03-02 at 12 59 18 am

**Pull request merged**